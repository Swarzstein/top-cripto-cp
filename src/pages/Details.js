import React from 'react';
import { useSelector } from 'react-redux';
import { NavLink, useParams } from 'react-router-dom';
import { FaChevronLeft, FaMicrophone, FaCog } from 'react-icons/fa';
// import store from '../redux/configureStore';
// import { getCrypto } from '../redux/currencies/currencies';

export default function Details() {
  const currencies = useSelector((state) => state.currencies);
  const { id } = useParams();
  const currency = currencies.find((currency) => currency.id === id);
  return (
    <>
      <header className="App-header">
        <nav>
          <div className="flex">
            <NavLink to="/"><h2><FaChevronLeft /></h2></NavLink>
            <h3>2023</h3>
          </div>
          <h3>Details</h3>
          <div className="icons-container">
            <FaMicrophone />
            <FaCog />
          </div>
        </nav>
      </header>
      <div className="currency-details">
        <div className="currency-head">
          <h2>{currency.symbol}</h2>
          <h2>{currency.name}</h2>
        </div>
        <div className="detail bg1">
          <h3>Price</h3>
          <h3>
            US$
            {currency.priceUsd}
          </h3>
        </div>
        <div className="detail bg2">
          <h3>Prev. Price</h3>
          <h3>
            US$
            {currency.vwap24Hr}
          </h3>
        </div>
        <div className="detail bg1">
          <h3>Change</h3>
          <h3>
            {currency.changePercent24Hr}
            %
          </h3>
        </div>
        <div className="detail bg2">
          <h3>Market Cap</h3>
          <h3>
            US$
            {currency.marketCapUsd.slice(0, (currency.marketCapUsd.indexOf('.') + 6))}
          </h3>
        </div>
        <div className="detail bg1">
          <h3>Supply</h3>
          <h3>
            {currency.symbol}
            {currency.supply.slice(0, (currency.supply.indexOf('.') + 6))}
          </h3>
        </div>
        <div className="detail bg2">
          <h3>Volume</h3>
          <h3>
            US$
            {currency.volumeUsd24Hr.slice(0, (currency.volumeUsd24Hr.indexOf('.') + 6))}
          </h3>
        </div>
        <div className="detail bg1">
          <h3>More</h3>
          <h4><a href={currency.explorer}>{currency.explorer}</a></h4>
        </div>
      </div>
    </>
  );
}
