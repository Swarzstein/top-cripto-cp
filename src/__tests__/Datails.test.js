import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import Details from '../pages/Details';
import currenciesReducerMock from '../utils/mockCurrenciesReducer';

describe('Currency component', () => {
  const store = configureStore({
    reducer: {
      currencies: currenciesReducerMock,
    },
  });

  it('Currency box renders', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/bitcoin']}>
            <Routes>
              <Route path="/:id" element={<Details />} />
            </Routes>
          </MemoryRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
