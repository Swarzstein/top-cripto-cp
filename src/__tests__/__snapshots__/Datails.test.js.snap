// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency component Currency box renders 1`] = `
Array [
  <header
    className="App-header"
  >
    <nav>
      <div
        className="flex"
      >
        <a
          className=""
          href="/"
          onClick={[Function]}
        >
          <h2>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
              />
            </svg>
          </h2>
        </a>
        <h3>
          2023
        </h3>
      </div>
      <h3>
        Details
      </h3>
      <div
        className="icons-container"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 352 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
          />
        </svg>
      </div>
    </nav>
  </header>,
  <div
    className="currency-details"
  >
    <div
      className="currency-head"
    >
      <h2>
        BTC
      </h2>
      <h2>
        Bitcoin
      </h2>
    </div>
    <div
      className="detail bg1"
    >
      <h3>
        Price
      </h3>
      <h3>
        US$
        22785.1563951154857609
      </h3>
    </div>
    <div
      className="detail bg2"
    >
      <h3>
        Prev. Price
      </h3>
      <h3>
        US$
        22298.7286729193235851
      </h3>
    </div>
    <div
      className="detail bg1"
    >
      <h3>
        Change
      </h3>
      <h3>
        3.3833553607951367
        %
      </h3>
    </div>
    <div
      className="detail bg2"
    >
      <h3>
        Market Cap
      </h3>
      <h3>
        US$
        439564105420.61628
      </h3>
    </div>
    <div
      className="detail bg1"
    >
      <h3>
        Supply
      </h3>
      <h3>
        BTC
        19291687.00000
      </h3>
    </div>
    <div
      className="detail bg2"
    >
      <h3>
        Volume
      </h3>
      <h3>
        US$
        11870305812.48817
      </h3>
    </div>
    <div
      className="detail bg1"
    >
      <h3>
        More
      </h3>
      <h4>
        <a
          href="https://blockchain.info/"
        >
          https://blockchain.info/
        </a>
      </h4>
    </div>
  </div>,
]
`;
